### PR TITLE
fix(tasks-new): Reset selector contacts on modal close

### PR DESCRIPTION
**Pull request by shahzadaziz**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @shahzadaziz has signed the CLA. Thanks for the pull request!

File: src/elements/content-sidebar/activity-feed/task-new/Task.js [hunk 1]
```diff
@@ -96,7 +97,12 @@ class Task extends React.Component<Props, State> {
     };
 
     handleModalClose = () => {
-        this.setState({ isEditing: false });
+        const { onModalClose } = this.props;
+        this.setState({ isEditing: false, modalError: null });
+
+        if (onModalClose) {
```
Comment (jstoffan) [inline src/elements/content-sidebar/activity-feed/task-new/Task.js]: Is a test necessary?

File: src/elements/content-sidebar/activity-feed/task-new/Task.js [hunk 2]
```diff
@@ -96,7 +97,12 @@ class Task extends React.Component<Props, State> {
     };
 
     handleModalClose = () => {
-        this.setState({ isEditing: false });
+        const { onModalClose } = this.props;
+        this.setState({ isEditing: false, modalError: null });
```
Comment (jstoffan) [inline src/elements/content-sidebar/activity-feed/task-new/Task.js]: The initial state for `modalError` is `undefined`. Does `null` here mean something different? If not, we should try to stay consistent and avoid `null`. Also, should the `modalError` be cleared in any other circumstances that may have been previously missed?

File: src/elements/content-sidebar/activity-feed/activity-feed/ActivityFeed.js [hunk 3]
```diff
@@ -31,6 +31,7 @@ type Props = {
     onTaskAssignmentUpdate?: Function,
     onTaskCreate?: Function,
     onTaskDelete?: Function,
+    onTaskModalClose?: Function,
```
Comment (priyajeet) [inline src/elements/content-sidebar/activity-feed/activity-feed/ActivityFeed.js]: Fine for now, but for new entries should start adding explicit flow types so () => void kind

Comment (alexkrolick) [reply src/elements/content-sidebar/activity-feed/activity-feed/ActivityFeed.js]: () => any is equivalent to Function, right?

File: src/elements/content-sidebar/activity-feed/task-new/Task.js (hunk 1 above)
Comment (shahzadaziz) [reply src/elements/content-sidebar/activity-feed/task-new/Task.js]: Added test

**Pull request merged**